scripts/inference/benchmarking/README.md

```diff
@@ -14,4 +18,118 @@ To run the benchmarking script, use the following command:
 
 `python benchmark.py yamls/1b.yaml`
 
-To run the scripts on [The MosaicML platform](https://www.mosaicml.com/blog/mosaicml-cloud-demo) we've also included scripts and associated `.yaml files` in the `mcli` folder.
+To run the scripts on [The MosaicML platform](https://www.mosaicml.com/blog/mosaicml-cloud-demo) we've also included scripts and associated `.yaml` files in the `mcli` folder.
+
+## LLM Inference Overview and Results
+
+### Background
+
+LLM inference consists of two stages: _prefill_ and _decode_. It's important to understand the difference between these stages as latency scales differently in each stage.
+
+During _prefill_, the model processes the input tokens/prompt/context. This is done in a single forward pass, making this stage fast, with excellent use of GPU hardware (ie. high Model Flop Utilization aka [MFU](https://github.com/mosaicml/llm-foundry/tree/main/scripts/train/benchmarking#mfu)). Typically, if people talk about LLM inference being slow, this is _not_ the stage that they are referring to.
+
+During _decode_, the model generates output tokens one at a time, ie. autoregressively. This requires making N forward passes of the model for N tokens. This stage is slow and inefficient, because it requires moving gigabytes of model weights and pre-filled values for every single forward pass. Here, latency scales (mostly) linearly with the number of output tokens. Why mostly linear? When generating long sequences, the overhead from the attention operation becomes visible, which is quadratic.
+
+##### KV cache
+
+One of the important elements of LLM inference is a Key-Value (KV) cache. As part of the prefill stage, the key and value tensors are computed and cached for each token in the prompt across the model layers. This is helpful because the initial prompt does not change during inference, which allows the model layers to simply re-use the cached values, rather than spending compute on re-computing the tokens for the context. \
+To set up the KV cache for benchmarking, set `use_cache: true` in the `.yaml` file.
+
+
+### Results
+Now that we have the necessary background to reason about inference, we can move onto the fun stuff, ie. real-world benchmark results.
+
+TODO: we will be releasing the configs used to reproduce these latency results soon!
+### I want to run generation with MPT-[X], how long does it take to process different size inputs?
+
+#### Setup
+We use a single A100 80GB for inference, running with precision `bf16` and the `torch` implementation of attention. We include the highest possible batch size that is able to produce 1000 output tokens without running out of GPU memory.
+
+Here we show how latency changes for a given input prompt length, while varying batch size and output length.
+This gives a rule of thumb of how fast you can expect MPT to be based on different generation parameters.
+#### TL;DR
```

Comment:
not sure what the purpose of saying tldr here is

Reply:
Should I say technical details? I wanted a succint way for people to see the experimental details

Reply:
"technical details" sounds good